locator/tablets.cc

```diff
@@ -240,6 +240,10 @@ class tablet_effective_replication_map : public effective_replication_map {
         return {get_endpoint_for_host_id(info->pending_replica.host)};
     }
 
+    virtual std::optional<inet_address_vector_replica_set> get_endpoints_for_reading(const token& search_token) const override {
+        return std::nullopt;
```

Comment:
Should be simple: if `info` is engaged and we're in `read_new` state, return `info.next`, otherwise `nullopt`.
cc @tgrabiec 
We can leave it for a follow-up.